### DISCUSSION TITLE
Issue with installing Foundry

### DISCUSSION THREAD

**Discussion by Divya-SC** [Cyfrin/foundry-full-course-cu#2623] [General]
I tried to install foundry on mac, but unable to install forge,cast,anvil, and chisel, I don't understand
Can anyone help me to sortout this issue
divyaarun@Divyas-Air ~ % curl -L https://foundry.paradigm.xyz | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   167  100   167    0     0    181      0 --:--:-- --:--:-- --:--:--   181
100  2189  100  2189    0     0   1574      0  0:00:01  0:00:01 --:--:--     0
Installing foundryup...

warning: libusb not found. You may need to install it manually on MacOS via Homebrew (brew install libusb).

Detected your preferred shell is zsh and added foundryup to PATH.
Run 'source /Users/divyaarun/.zshenv' or start a new terminal session to use foundryup.
Then, simply run 'foundryup' to install Foundry.
divyaarun@Divyas-Air ~ % source /Users/divyaarun/.zshenv
divyaarun@Divyas-Air ~ % foundryup

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

 ╔═╗ ╔═╗ ╦ ╦ ╔╗╔ ╔╦╗ ╦═╗ ╦ ╦         Portable and modular toolkit
 ╠╣  ║ ║ ║ ║ ║║║  ║║ ╠╦╝ ╚╦╝    for Ethereum Application Development
 ╚   ╚═╝ ╚═╝ ╝╚╝ ═╩╝ ╩╚═  ╩                 written in Rust.

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

Repo       : https://github.com/foundry-rs/
Book       : https://book.getfoundry.sh/
Chat       : https://t.me/foundry_rs/
Support    : https://t.me/foundry_support/
Contribute : https://github.com/orgs/foundry-rs/projects/2/

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

foundryup: installing foundry (version nightly, tag nightly)
foundryup: downloading latest forge, cast, anvil, and chisel
########################################################################################################################################################################################### 100.0%
foundryup: downloading manpages
########################################################################################################################################################################################### 100.0%
**dyld[2660]: Library not loaded: '/usr/local/opt/libusb/lib/libusb-1.0.0.dylib'
  Referenced from: '/Users/divyaarun/.foundry/bin/forge'
  Reason: tried: '/usr/local/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/usr/local/lib/libusb-1.0.0.dylib' (no such file), '/usr/lib/libusb-1.0.0.dylib' (no such file)
foundryup: command failed: /Users/divyaarun/.foundry/bin/forge --version
foundryup: installed - 
dyld[2665]: Library not loaded: '/usr/local/opt/libusb/lib/libusb-1.0.0.dylib'
  Referenced from: '/Users/divyaarun/.foundry/bin/cast'
  Reason: tried: '/usr/local/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/usr/local/lib/libusb-1.0.0.dylib' (no such file), '/usr/lib/libusb-1.0.0.dylib' (no such file)
foundryup: command failed: /Users/divyaarun/.foundry/bin/cast --version
foundryup: installed - 
dyld[2668]: Library not loaded: '/usr/local/opt/libusb/lib/libusb-1.0.0.dylib'
  Referenced from: '/Users/divyaarun/.foundry/bin/anvil'
  Reason: tried: '/usr/local/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/usr/local/lib/libusb-1.0.0.dylib' (no such file), '/usr/lib/libusb-1.0.0.dylib' (no such file)
foundryup: command failed: /Users/divyaarun/.foundry/bin/anvil --version
foundryup: installed - 
dyld[2671]: Library not loaded: '/usr/local/opt/libusb/lib/libusb-1.0.0.dylib'
  Referenced from: '/Users/divyaarun/.foundry/bin/chisel'
  Reason: tried: '/usr/local/opt/libusb/lib/libusb-1.0.0.dylib' (no such file), '/usr/local/lib/libusb-1.0.0.dylib' (no such file), '/usr/lib/libusb-1.0.0.dylib' (no such file)
foundryup: command failed: /Users/divyaarun/.foundry/bin/chisel --version**
foundryup: installed - 
foundryup: done!
divyaarun@Divyas-Air ~ %

**Comment by EngrPips** [comment]
Hello @Divya-SC, I think you need to install `libusb`, Please read this Ethereum-stack-exchange post https://stackoverflow.com/questions/3853634/how-to-set-up-libusb-on-mac-os-x I believe it will help.
